src/wallet/wallet.cpp

```diff
@@ -1180,9 +1181,22 @@ void CWallet::ResendWalletTransactions(bool fForce)
         LOCK(cs_wallet);
         // Sort them in chronological order
         multimap<unsigned int, CWalletTx*> mapSorted;
+        std::vector<CWalletTx> to_be_erased;
         for (auto &item : mapWallet)
         {
             CWalletTx& wtx = item.second;
+
+            AssertLockHeld(cs_main);
+
+            if (wtx.vContracts[0].m_type == GRC::ContractType::MRC) {
+                GRC::MRC mrc = *(wtx.vContracts[0].SharePayloadAs<GRC::MRC>());
+
+                // Remove MRC transaction if it went stale.
+                if (mrc.m_last_block_hash != hashBestChain) {
+                    to_be_erased.push_back(wtx);
+                }
+            }
```

Comment:
This is good. Should we also put a staleness check in the ConnectBlock call so that the user doesn't have to wait 5 minutes + the next block staking latency to have the MRC transaction absorbed so they can retry? (Because they will not be able to add a new MRC until the old one is cleared out.)